test/strtoultest.c

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2016-2024 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/crypto.h>
+#include <internal/nelem.h>
+#include "testutil.h"
+
+struct strtoul_test_entry {
+    char *input; /* the input string */
+    int base; /* the base we are converting in */
+    unsigned long expect_val; /* the expected value we should get */
+    int expect_err;  /* the expected error we expect to recieve */
+    size_t expect_endptr_offset; /* the expected endptr offset, +1 for NULL */
+};
+
+static struct strtoul_test_entry strtoul_tests[] = {
```

Comment:
while the current implementation uses system `strtoul`, it's not specified that way, so we should check inputs near, at, and beyond `ULONG_MAX` (and negative inputs, too - this behaviour of system `strtoul` may not be what we want)